core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java

```diff
@@ -192,15 +193,24 @@ public final Object visitClearCommand(InvocationContext ctx, ClearCommand comman
                return;
             }
          }
-         // If this node has topology higher than some of the nodes and the nodes could not respond
-         // with the remote entry, these nodes are blocking the response and therefore we can get only timeouts.
-         // Therefore, if we got here it means that we have lower topology than some other nodes and we can wait
-         // for it in StateTransferInterceptor and retry the read later.
-         // TODO: These situations won't happen as soon as we'll implement 4-phase topology change in ISPN-5021
-         throw new OutdatedTopologyException("Did not get any successful response, got " + responses);
+         throw handleMissingSuccessfulResponse(responses);
       });
    }
 
+   protected static CacheException handleMissingSuccessfulResponse(Map<Address, Response> responses) {
+      if (responses.values().stream().anyMatch(UnsureResponse.class::isInstance)) {
+         // We got only unsure responses, as all nodes that were read-owners at the time when we've sent
+         // the request have progressed to newer topology. However we are guaranteed to have progressed
+         // to a topology at most one older, and can immediately retry.
+         return OutdatedTopologyException.getCachedInstance();
+      } else {
+         // Another instance when we don't get any successful response is when all owners are lost. We'll handle
+         // this later in StateTransferInterceptor, as we have to signal this to PartitionHandlingInterceptor
+         // if that's present.
+         return AllOwnersLostException.INSTANCE;
```

Comment:
> If there was an exception, it would be thrown somewhere from JGroupsTransport, this code won't handle any ExceptionResponse instances.

Fair enough, but it's not very clear from the code.

> [...] I would rather expose OTE.INSTANCE directly [...]

I'm also ok with removing the delegation method from OTE, I just like code that does the same thing to look the same :)

> [...] checking "Why the hell did I get this exception" sounds prone to races to me [...]

I think PHI is still prone to races, if the partition check succeeds it will throw the `AOLE` back to the user instead of an `AvailabilityException`. And we don't want the user to see `AOLE` any more then we want them to see `OutdatedTopologyException`... even with partition handling disabled, the user should see a `null` value instead of an exception.

IMO `OTE` shouldn't be so surprising in a degraded cache, because read requests are sent based on the (outdated, sort of) stable topology's CH. I agree `AOLE` makes things a bit clearer, but ignoring the exception name, the implementation should be the same.